skyplane/api/api.py

```diff
@@ -0,0 +1,262 @@
+            num_connections=config_default_flags.get("num_connections"),
+            max_instances=max_instances,
+            solver_total_gbyte_to_transfer=sum(src_obj.size for src_obj, _ in transfer_pairs) if solve else None,
+            solver_target_tput_per_vm_gbits=solver_target_tput_per_vm_gbits,
+            solver_throughput_grid=solver_throughput_grid,
+            solver_verbose=solver_verbose,
+            args=args,
+        )
+        job = ReplicationJob(
+            source_region=topo.source_region(),
+            source_bucket=bucket_src,
+            dest_region=topo.sink_region(),
+            dest_bucket=bucket_dst,
+            transfer_pairs=transfer_pairs,
+        )
+        confirm_transfer(topo=topo, job=job, ask_to_confirm_transfer=not confirm)
+
+        small_transfer_cmd = replicate_small_cp_cmd(src, dst, recursive)
+        if (
+            config_default_flags.get("native_cmd_enabled")
+            and (job.transfer_size / GB) < config_default_flags.get("native_cmd_threshold_gb")
+            and small_transfer_cmd
+        ):
+            # typer.secho(f"Transfer is small enough to delegate to native tools. Delegating to: {small_transfer_cmd}", fg="yellow")
+            os.system(small_transfer_cmd)
+            return 0
+        else:
+            transfer_stats = launch_replication_job(
+                topo=topo,
+                job=job,
+                debug=debug,
+                reuse_gateways=reuse_gateways,
+                use_bbr=config_default_flags.get("bbr"),
+                use_compression=config_default_flags.get("compress") if src_region_tag != dst_region_tag else False,
+                use_e2ee=config_default_flags.get("encrypt_e2e") if src_region_tag != dst_region_tag else False,
+                use_socket_tls=config_default_flags.get("encrypt_socket_tls") if src_region_tag != dst_region_tag else False,
+                aws_instance_class=config_default_flags.get("aws_instance_class"),
+                aws_use_spot_instances=config_default_flags.get("aws_use_spot_instances"),
+                azure_instance_class=config_default_flags.get("azure_instance_class"),
+                azure_use_spot_instances=config_default_flags.get("azure_use_spot_instances"),
+                gcp_instance_class=config_default_flags.get("gcp_instance_class"),
+                gcp_use_premium_network=config_default_flags.get("gcp_use_premium_network"),
+                gcp_use_spot_instances=config_default_flags.get("gcp_use_spot_instances"),
+                multipart_enabled=multipart,
+                multipart_min_threshold_mb=config_default_flags.get("multipart_min_threshold_mb"),
+                multipart_chunk_size_mb=config_default_flags.get("multipart_chunk_size_mb"),
+                multipart_max_chunks=config_default_flags.get("multipart_max_chunks"),
+                error_reporting_args=args,
+            )
+            if config_default_flags.get("verify_checksums"):
```

Comment:
Exceptions should be raised directly without catching them, and content shouldn't be printed out for API calls